src/features/defi/providers/osmosis/components/OsmosisManager/Deposit/components/Deposit.tsx

```diff
@@ -0,0 +1,375 @@
+       * for the provided liquidity
+       */
+      const shareOutAmountBaseUnits = bnOrZero(poolTotalSharesBaseUnits)
+        .multipliedBy(
+          bnOrZero(inputAssetAmountBaseUnits).dividedBy(bnOrZero(poolAssetAmountBaseUnits)),
+        )
+        .toFixed(0)
+        .toString()
+
+      ret.allocationFraction = allocationFraction
+      ret.shareOutAmount = shareOutAmountBaseUnits
+      return ret
+    },
+    [state],
+  )
+
+  const handleContinue = useCallback(
+    async (formValues: DepositValues) => {
+      if (
+        !(
+          state &&
+          contextDispatch &&
+          userAddress &&
+          opportunity &&
+          underlyingAsset0 &&
+          underlyingAsset1
+        )
+      )
+        return
+
+      const asset0AmountBaseUnits = bnOrZero(formValues.cryptoAmount1)
+        .pow(10, bnOrZero(underlyingAsset0.precision))
+        .toString()
+
+      const asset1AmountBaseUnits = bnOrZero(formValues.cryptoAmount2)
+        .pow(10, bnOrZero(underlyingAsset1.precision))
+        .toString()
+
+      // set deposit state for future use
+      contextDispatch({
+        type: OsmosisDepositActionType.SET_DEPOSIT,
+        payload: {
+          underlyingAsset0: {
+            amount: asset0AmountBaseUnits,
+            denom: fromAssetId(underlyingAsset0.assetId).assetReference,
+          },
+          underlyingAsset1: {
+            amount: asset1AmountBaseUnits,
+            denom: fromAssetId(underlyingAsset1.assetId).assetReference,
+          },
+          shareOutAmount: calculateAllocations(underlyingAsset0, formValues.cryptoAmount1)
+            .shareOutAmount,
+        },
+      })
+      contextDispatch({ type: OsmosisDepositActionType.SET_LOADING, payload: true })
+      try {
+        const estimatedFeeCrypto = await getDepositFeeEstimate()
+        if (!estimatedFeeCrypto) return
+        contextDispatch({
+          type: OsmosisDepositActionType.SET_DEPOSIT,
+          payload: { estimatedFeeCrypto },
+        })
+        onNext(DefiStep.Confirm)
+        contextDispatch({ type: OsmosisDepositActionType.SET_LOADING, payload: false })
+      } catch (error) {
+        moduleLogger.error({ fn: 'handleContinue', error }, 'Error on continue')
+        toast({
+          position: 'top-right',
+          description: translate('common.somethingWentWrongBody'),
+          title: translate('common.somethingWentWrong'),
+          status: 'error',
+        })
+        contextDispatch({ type: OsmosisDepositActionType.SET_LOADING, payload: false })
+      }
+    },
+    [
+      state,
+      contextDispatch,
+      userAddress,
+      opportunity,
+      underlyingAsset0,
+      underlyingAsset1,
+      calculateAllocations,
+      getDepositFeeEstimate,
+      onNext,
+      toast,
+      translate,
+    ],
+  )
+
+  const handleCancel = useCallback(() => {
+    browserHistory.goBack()
+  }, [browserHistory])
```

Comment:
Purely OCD but if you want to avoid one stack trace

```suggestion
  const handleCancel = browserHistory.goBack()
```

No need to have any deps or memoize anything since `browserHistory` itself is coming from a hook and `goBack` is referentially stable